src/Autosuggest.js

```diff
@@ -222,8 +238,9 @@ class Autosuggest extends Component {
               }
             } else if (suggestions.length > 0) {
               const { newFocusedSectionIndex, newFocusedItemIndex } = data;
+              const safeValue = valueBeforeUpDown ? valueBeforeUpDown : value;
               const newValue = newFocusedItemIndex === null ?
-                valueBeforeUpDown :
+                safeValue :
```

Comment:
Great, thanks for the detailed explanation and for fixing this bug!
Can I just suggest that we make the code a little more self explanatory:
```js
const safeValue = valueBeforeUpDown === null // This can happen if, for example, user hovers on the first suggestion and then clicks Up
  ? value 
  : valueBeforeUpDown;
```